layers/timeline_semaphore.c

```diff
@@ -308,6 +308,12 @@ static struct object_map global_map = {
     .lock = PTHREAD_MUTEX_INITIALIZER,
 };
 
+/* Mapping of physical device object */
+
+static struct object_map global_physical_device_map = {
```

Comment:
Removing the physical device entries at instance destroy sounds simpler.

Reply:
Alright, let me send PR change by change.

Reply:
> 
> 
> > I meant create info is const, we need to clone the create info and remove the timelineSemaphore feaure from the create info so that ICD creates device successfully.
> 
> Oh right... I keep testing this on a version of our implementation that already supports the extension...

@djdeath, do you have time to help on this? I'm afraid I will be very busy soon.